docs/source/pytest.rst

```diff
@@ -0,0 +1,358 @@
+* reporting all executed cocotb tests.
+* parallel execution of cococtb runners by using `pytest-xdist`_ plugin
+
+Enabling Plugin
+===============
+
+:py:mod:`cocotb_tools.pytest.plugin` can be enabled in various ways.
+
+When using `pyproject.toml`_ file (recommended way):
+
+.. code:: toml
+
+   [project.entry-points.pytest11]
+   cocotb = "cocotb_tools.pytest.plugin"
+
+When using ``pytest.ini`` file:
+
+.. code:: ini
+
+   [pytest]
+   addopts = -p cocotb_tools.pytest.plugin
+
+When using ``setup.cfg`` file:
+
+.. code:: ini
+
+   [options.entry_points]
+   pytest11 =
+     cocotb = cocotb_tools.pytest.plugin
+
+When using ``setup.py`` file:
+
+.. code:: python
+
+   from setuptools import setup
+
+   setup(
+       # ...,
+       entry_points={
+           "pytest11": [
+               "cocotb = cocotb_tools.pytest.plugin",
+           ],
+       },
+   )
+
+By defining the global variable ``pytest_plugins`` when using ``conftest.py`` file
+(it must be located in the root of the project):
+
+.. code:: python
+
+   pytest_plugins = ("cocotb_tools.pytest.plugin",)
+
+By defining the ``PYTEST_PLUGINS`` environment variable:
+
+.. code:: shell
+
+   export PYTEST_PLUGINS="cocotb_tools.pytest.plugin"
+
+By using ``-p <plugin>`` option when invoking `pytest`_ command line interface:
+
+.. code:: shell
+
+   pytest -p cocotb_tools.pytest.plugin ...
+
+Building and Testing
+====================
+
+:py:class:`cocotb_tools.pytest.hdl.HDL` wraps :py:class:`cocotb_tools.runner.Runner`
+allowing to fully configure cocotb runner by using command line arguments ``--cocotb-*``,
+configuration files like `pyproject.toml`_ or `fixture`_ arguments.
+
+Plugin includes own ``hdl`` fixture that will create new instance of :py:class:`cocotb_tools.pytest.hdl.HDL`
+with `pytest`_ built-in `request`_ fixture that is providing information of the requesting test function
+including current configuration of `pytest`_.
+
+Example content of ``conftest.py`` file:
+
+.. code:: python
+
+   import pytest
+   from cocotb_tools.pytest.hdl import HDL
+
+
+   @pytest.fixture(name="sample_module")
+   def sample_module_fixture(hdl: HDL) -> HDL:
+       """Define HDL design by adding HDL source files to it.
+
+       Args:
+           hdl: Fixture created by cocotb plugin, representing HDL design.
+
+       Returns:
+           Representation of HDL design with added HDL source files.
+       """
+       hdl.sources = (
+           # List HDL source files,
+           "sample_module.sv",
+       )
+
+       return hdl
+
+
+Example content of ``test_sample_module.py`` file:
+
+.. code:: python
+
+   import pytest
+   from cocotb_tools.pytest.hdl import HDL
+
+
+   # Without providing positional arguments or test_module option to cocotb decorator,
+   # plugin will use current file as cocotb testbench (Python file with cocotb tests).
+   # If toplevel option was not provided, it will be based on name of first test_module
+   # but with removed test_* prefix or *_test suffix.
```

Comment:
Maybe this should be a separate paragraph? Let the examples be examples: short and self-explanatory *executable* code as possible.